actions/generate-schedule/scheduler/scheduler.go

```diff
@@ -33,70 +34,64 @@ func NewScheduler(userSource users.Source, shiftDurationDays int) (*Scheduler, e
 	}, nil
 }
 
-// Schedule creates a new Schedule from start (inclusive) to stop (exclusive). Will return a 1-entry schedule if
-// stop is before start.
+// Schedule creates a new Schedule that includes whole shifts of `Scheduler.shiftDuration` from start (inclusive) to
+// stop (inclusive).  Will return an error if stop is before start, or either start are stop are zero values.
 func (s *Scheduler) Schedule(start, stop time.Time) (*schedule.Schedule, error) {
-	sched := &schedule.Schedule{
-		Shifts: []*schedule.Shift{
-			{
-				StartDate: start.Format(DateFormat),
-				User:      s.userSource.NextUser(),
-			},
-		},
+	if start.IsZero() || stop.IsZero() {
+		return nil, fmt.Errorf("neither start (%v) nor stop (%v) can be zero values", start, stop)
+	}
+
+	if stop.Before(start) {
```

Comment:
if they're equal that's also an issue, right? So `if !stop.After(start)`?

Reply:
I changed `stop` to be inclusive, so they can in fact be equal. 

```
go run ./actions/generate-schedule/main.go --start 2020-06-01 --stop 2020-06-01 --shiftDurationDays 1 --users ttomsu --out foobar.yml
```
is a valid. 

In investigating this, I found several more bugs in tests.